totalRP3/modules/dashboard/dashboard.lua

```diff
@@ -146,6 +146,98 @@ local ShowingHelm, ShowingCloak, ShowCloak, ShowHelm = ShowingHelm, ShowingCloak
 local Button_Cape, Button_Helmet, Button_Status, Button_RPStatus;
 local SendChatMessage, UnitIsDND, UnitIsAFK = SendChatMessage, UnitIsDND, UnitIsAFK;
 
+--- Formats a list of names for display in the About text. Each line is placed
+--  onto a separate line with a prefixing list indicator ("- ").
+--
+--  If a role is given it will be included in each name.
+local function getFormattedAboutCreditsMembers(credits, role)
+	-- No credits? We don't want to display a "- " for no reason.
+	if not credits or #credits == 0 then
+		return "";
+	end
+
+	if role then
+		-- Localise the role and then iterate over the credits, copying
+		-- each entry and including the role information.
+		local localisedRole = loc("THANK_YOU_ROLE_" .. role);
+		local creditsWithRole = Utils.table.getTempTable();
+
+		for i = 1, #credits do
+			local name = credits[i];
+			creditsWithRole[i] = loc("THANK_YOU_WITH_ROLE"):format(name, localisedRole);
+		end
+
+		credits = creditsWithRole;
+	end
+
+	-- Net result here is we'll insert a newline and a "- " between each
+	-- item in the list. We'll prefix it with a "- " in the format call so
+	-- the first item is properly indented too.
+	local formattedMembers = ("- %s"):format(table.concat(credits, "\n- "));
+
+	-- If there's a role then credits points to a temp table still.
+	if role then
+		Utils.table.releaseTempTable(credits);
+	end
+
+	return formattedMembers;
+end
+
+--- Formats a given credits group into a single string. The group can contain
+--  multiple roles inside of it, in which case all members of the associated
+--  roles will be joined together into a single newline-separated list.
+--
+--  If multiple roles are present in the group, role information is attached
+--  to the names of the members, otherwise it is omitted.
+local function getFormattedAboutCreditsGroup(group)
+	-- If the group only has one role, things are simple - we're only going
+	-- to display one lists worth of names.
+	if #group <= 1 then
+		local role = group[1];
+		local members = Globals.ADDON_CREDITS[role];
+		return getFormattedAboutCreditsMembers(members);
+	end
+
+	-- You've got a few roles, that's fine. Bit more effort.
+	local groupMembers = Utils.table.getTempTable();
+	for i = 1, #group do
+		local role = group[i];
+		local members = Globals.ADDON_CREDITS[role];
+
+		-- We want to include the role information when there's a single
+		-- logical group with multiple roles attached.
+		table.insert(groupMembers, getFormattedAboutCreditsMembers(members, role));

```

Comment:
Import `table.insert` at the top of the file, to get a local reference and spare a few global lookups. Especially here since we are in a loop of unknown size.